instrumentation/jmx-metrics/javaagent/src/main/java/io/opentelemetry/instrumentation/javaagent/jmx/JmxMetricInsightInstaller.java

```diff
@@ -5,38 +5,43 @@
 
 package io.opentelemetry.instrumentation.javaagent.jmx;
 
-import static java.util.logging.Level.FINE;
-import static java.util.logging.Level.INFO;
-
 import com.google.auto.service.AutoService;
 import io.opentelemetry.api.GlobalOpenTelemetry;
-import io.opentelemetry.instrumentation.jmx.engine.JmxMetricInsight;
-import io.opentelemetry.instrumentation.jmx.engine.MetricConfiguration;
-import io.opentelemetry.instrumentation.jmx.yaml.RuleParser;
+import io.opentelemetry.instrumentation.jmx.JmxTelemetry;
+import io.opentelemetry.instrumentation.jmx.JmxTelemetryBuilder;
 import io.opentelemetry.javaagent.extension.AgentListener;
 import io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdk;
 import io.opentelemetry.sdk.autoconfigure.internal.AutoConfigureUtil;
 import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
-import java.io.InputStream;
-import java.nio.file.Files;
 import java.nio.file.Paths;
 import java.time.Duration;
-import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
 
 /** An {@link AgentListener} that enables JMX metrics during agent startup. */
 @AutoService(AgentListener.class)
 public class JmxMetricInsightInstaller implements AgentListener {
 
+  private static final Logger logger = Logger.getLogger(JmxMetricInsightInstaller.class.getName());
+
   @Override
   public void afterAgent(AutoConfiguredOpenTelemetrySdk autoConfiguredSdk) {
     ConfigProperties config = AutoConfigureUtil.getConfig(autoConfiguredSdk);
 
     if (config.getBoolean("otel.jmx.enabled", true)) {
-      JmxMetricInsight service =
-          JmxMetricInsight.createService(
-              GlobalOpenTelemetry.get(), beanDiscoveryDelay(config).toMillis());
-      MetricConfiguration conf = buildMetricConfiguration(config);
-      service.startLocal(conf);
+      JmxTelemetryBuilder jmx =
+          JmxTelemetry.builder(GlobalOpenTelemetry.get())
+              .beanDiscoveryDelay(beanDiscoveryDelay(config).toMillis());
+
+      try {
+        config.getList("otel.jmx.config").stream().map(Paths::get).forEach(jmx::addCustomRules);
+        config.getList("otel.jmx.target.system").forEach(jmx::addClassPathRules);
+      } catch (IllegalArgumentException e) {
+        // for now only log JMX errors as they do not prevent agent startup
+        logger.log(Level.SEVERE, "Error while loading JMX configuration", e);
```

Comment:
[for reviewer] for now an invalid configuration does not prevent the agent from starting, it just logs an error which was also the current behavior.

Reply:
The overall behavior does not change here, the agent currently starts with warnings in the logs when there are parsing errors, now an exception is thrown and it only logs the first error thanks to this catch statement.